### ISSUE TITLE
`no-restricted-globals` - Allow `location.origin` and more without `globalThis`

### ISSUE THREAD

**Issue by fregante** [xojs/eslint-config-xo#115]
I'm not sure what the final decision here is, I upgraded to the latest eslint-config-xo (unicorn v67) and it's marking lines previously fixed by this rule as `Unexpected use of 'location' no-restricted-globals`

```js
const actionUrl = new URL('search', location.origin);
```

_Originally posted by @fregante in https://github.com/sindresorhus/eslint-plugin-unicorn/issues/3161#issuecomment-4756660250_

**Comment by fregante** [comment]
Originally `no-restricted-globals` was used to block confusing globals like `event`, `name`, `addEventListener` and `innerWidth`, but it was recently altered to force `globalThis` on other globals that are largely known and used forever as globals, like `location`, `screen`, `history`, `confirm`, etc

I wouldn't be totally against just formalizing `globalThis.location`, it's just that it goes against what it has always been used. Not sure why `location` should be `globalThis.location` but `document`, `chrome` and `console` are good as is, for example.
